state/protocol/badger/snapshot.go

```diff
@@ -270,70 +270,70 @@ func (q *EpochQuery) Current() protocol.Epoch {
 	if err != nil {
 		return NewInvalidEpoch(err)
 	}
-	return NewCommittedEpoch(setup, commit)
+	return q.NewCommittedEpoch(setup, commit)
 }
 
-// Next returns the next epoch.
+// Next returns the next epoch, if it is available.
 func (q *EpochQuery) Next() protocol.Epoch {
+
 	status, err := q.snap.state.epoch.statuses.ByBlockID(q.snap.blockID)
 	if err != nil {
 		return NewInvalidEpoch(err)
 	}
-	setup, err := q.snap.state.epoch.setups.ByID(status.CurrentEpoch.SetupID)
+	phase, err := status.Phase()
 	if err != nil {
 		return NewInvalidEpoch(err)
 	}
-	return q.ByCounter(setup.Counter + 1)
-}
+	// if we are in the staking phase, the next epoch is not setup yet
+	if phase == flow.EpochPhaseStaking {
+		return NewInvalidEpoch(protocol.ErrNextEpochNotSetup)
+	}
 
-// ByCounter returns the epoch with the given counter.
-func (q *EpochQuery) ByCounter(counter uint64) protocol.Epoch {
+	nextSetup, err := q.snap.state.epoch.setups.ByID(status.NextEpoch.SetupID)
+	if err != nil {
+		return NewInvalidEpoch(fmt.Errorf("failed to retrieve setup event for next epoch: %w", err))
+	}
+	if status.NextEpoch.CommitID == flow.ZeroID {
+		return q.NewSetupEpoch(nextSetup)
+	}
```

Comment:
would suggest to either consistently work with Epoch Phases or checking for `flow.ZeroID` 

```suggestion
	// if we are in the staking phase, the next epoch is not setup yet
	if phase == flow.EpochPhaseSetup {
		return q.NewSetupEpoch(nextSetup)
	}
```

Reply:
done